code/game/objects/machinery/portable_reagent_tank.dm

```diff
@@ -0,0 +1,214 @@
+	. += "[span_bold("Drag")] to yourself to undeploy."
+
+/obj/machinery/deployable/reagent_tank/update_icon()
+	//Remove overlays and reset the icon
+	overlays.Cut()
+	icon_state = initial(internal_item.icon_state)
+	//If reagents are present, add some overlays
+	if(reagents?.total_volume)
+		var/image/filling = image(icon, src, "[icon_state]")
+		var/percent = round((reagents.total_volume/max_volume) * 100)
+		switch(percent)
+			if(0 to 19)
+				filling.icon_state = "[icon_state]_1"
+			if(20 to 39)
+				filling.icon_state = "[icon_state]_2"
+			if(40 to 59)
+				filling.icon_state = "[icon_state]_3"
+			if(60 to 79)
+				filling.icon_state = "[icon_state]_4"
+			if(80 to 99)
+				filling.icon_state = "[icon_state]_5"
+			if(99 to INFINITY)
+				filling.icon_state = "[icon_state]_full"
+		filling.color = mix_color_from_reagents(reagents.reagent_list)
+		overlays += filling
+	//Change the sprite to the one that looks opened to indicate it is in refilling mode
+	if(is_refillable())
+		icon_state += "_open"
+
+/obj/machinery/deployable/reagent_tank/on_reagent_change()
+	update_icon()
+
+/obj/machinery/deployable/reagent_tank/attack_hand(mob/living/user)
+	. = ..()
+	if(user.a_intent == INTENT_GRAB)
+		return drink_from_nozzle(user)
+	var/obj/item/storage/internal_bag = internal_item
+	internal_bag.open(user)
+
+/obj/machinery/deployable/reagent_tank/attack_alien(mob/living/carbon/xenomorph/X, damage_amount, damage_type, damage_flag, effects, armor_penetration, isrightclick)
+	if(X.a_intent != INTENT_HARM)
+		return drink_from_nozzle(X, TRUE)
+	. = ..()
+
+///Process for drinking reagents directly from the dispenser's nozzle
+/obj/machinery/deployable/reagent_tank/proc/drink_from_nozzle(mob/living/user, is_xeno = FALSE)
+	if(isrobot(user) || issynth(user))
+		balloon_alert(user, "You are incapable of drinking!")
+		return FALSE
+	if(reagents?.total_volume)
+		if(!is_xeno)
+			//Everyone will be made aware of your nasty habits!
+			visible_message(span_alert("[user] is putting [user.p_their()] mouth on [src]'s nozzle. Gross!"))
+		if(!do_after(user, 0.5 SECONDS, FALSE, src, BUSY_ICON_DANGER))
+			return FALSE
+		if(is_xeno)
+			visible_message(span_alert("[user] sips from [src]'s nozzle. Adorable."))
+		playsound(user.loc,'sound/items/drink.ogg', 25, 2)
+		reagents.reaction(user, INGEST)
+		reagents.trans_to(user, 5)
+		return TRUE
+	balloon_alert(user, "It's empty!")
+
+/obj/machinery/deployable/reagent_tank/attackby(obj/item/I, mob/user, params)
+	if(I.is_refillable())
+		return FALSE //Beakers run the refilling code in afterattack()
+	return ..()
+
+/obj/machinery/deployable/reagent_tank/AltClick(mob/user)
+	if(is_refillable())
+		balloon_alert(user, "Dispense mode!")
+		DISABLE_BITFIELD(reagents.reagent_flags, REFILLABLE)
+		ENABLE_BITFIELD(reagents.reagent_flags, DRAINABLE)
+	else
+		balloon_alert(user, "Refill mode!")
+		DISABLE_BITFIELD(reagents.reagent_flags, DRAINABLE)
+		ENABLE_BITFIELD(reagents.reagent_flags, REFILLABLE)
```

Comment:
Just use |= and &= ~tbh, the defines are dumdum

Reply:
I think the defines are nice, easier for my small brain to understand. :(